lib/screens/explore_screens/regions_screens/edit_region/edit_region_image.dart

```diff
@@ -24,37 +24,32 @@ class EditRegionImage extends StatelessWidget {
     return BlocProvider(
       create: (_) => EditRegionBloc(region!),
       child: BlocConsumer<EditRegionBloc, EditRegionState>(
-        listenWhen: (previous, current) => current.pageCommand != null,
+        listenWhen: (_, current) => current.pageCommand != null,
         listener: (context, state) {
-          if (state.pageCommand != null) {
-            final pageCommand = state.pageCommand;
-
-            //need  a page command for this screen
-            if (pageCommand is RemoveAuthenticationScreen) {
-              // This pop remove the authentication screen
-              Navigator.of(context).pop();
-            } else if (pageCommand is ShowErrorMessage) {
-              eventBus.fire(ShowSnackBar(pageCommand.message));
-            } else if (pageCommand is NavigateToRoute) {
-              NavigationService.of(context).pushAndRemoveUntil(route: pageCommand.route, from: Routes.app);
-            } else if (pageCommand is EditRegionImage) {
-              BlocProvider.of<EditRegionBloc>(context).add(const OnEditRegionImage());
-            }
```

Comment:
Not working because it detect the Widget type insted the PageCommand type due the same name.

Reply:
I think this is why it broke, I dont think we should add the logic to the bloc.

Reply:
??

Reply:
![image](https://user-images.githubusercontent.com/51983923/167512234-f4c0daec-802d-4e59-87aa-a4cfb4855c78.png)